lib/fs/basicfs_watch.go

```diff
@@ -0,0 +1,95 @@
+// Copyright (C) 2016 The Syncthing Authors.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at http://mozilla.org/MPL/2.0/.
+
+// +build !solaris,!darwin solaris,cgo darwin,cgo
+
+package fs
+
+import (
+	"context"
+	"errors"
+	"path/filepath"
+	"strings"
+
+	"github.com/zillode/notify"
+)
+
+// Notify does not block on sending to channel, so the channel must be buffered.
+// The actual number is magic.
+const backendBuffer = 500
+
+func (f *BasicFilesystem) Watch(name string, ignore Matcher, ctx context.Context, ignorePerms bool) (<-chan Event, error) {
+	absName := filepath.Join(f.root, name)
```

Comment:
f.rooted?

Reply:
That messed up the tests, because the return value is not necessary filepath clean while the paths from the notify library are. I don't know anymore what exactly was the problem, I can run the tests with f.rooted again if you think this might be relevant for the filesystem as a whole.

Reply:
Well it would be nice to have only one way to do this.
I guess in the tests it could complain about the path not being absolute or something like that, but we somehow manage to work around it else where, so I'd like to know what the issue is.

Reply:
The problem is the trailing path separator added by `f.rooted`, which makes the `HasPrefix` check fail in all cases, causing the "processing a change outside" panic. I could use `f.rooted` and strip a trailing path separator instead of using `filepath.Join`.

Reply:
Sounds like a better option.

Reply:
Also, perhaps it's easier to add a trailing slash to the abs check, rather than strip?

Reply:
I'll change it and add a comment. What is the reason in the first place to not follow the `filepath` format, i.e. not having a trailing separator?

Reply:
Because of:
```
strings.HasPrefix('/root/myfolder', '/root/myfolder/x') == true
strings.HasPrefix('/root/myfolder', '/root/myfolder-that-did-not-intend-to-share/') == true
```